server/application-server/src/main/java/de/tum/in/www1/hephaestus/scheduler/GitHubDataSyncScheduler.java

```diff
@@ -26,30 +24,33 @@ public GitHubDataSyncScheduler(GitHubDataSyncService dataSyncService) {
         this.dataSyncService = dataSyncService;
     }
 
-    @PostConstruct
-    public void onStartup() {
+    @Override
+    public void run(ApplicationArguments args) throws Exception {
         if (runOnStartup) {
             logger.info("Starting initial GitHub data sync for Hephaestus...");
-            syncAllRepositories();
-            logger.info("Initial GitHub data sync completed successfully.");
+            syncData();
+            logger.info("Initial GitHub data sync completed.");
         }
     }
 
     @Scheduled(cron = "${monitoring.repository-sync-cron}")
     public void syncDataCron() {
-        logger.info("Starting daily GitHub data sync...");
-        syncAllRepositories();
-        logger.info("Daily GitHub data sync completed successfully.");
+        logger.info("Starting scheduled GitHub data sync...");
+        syncData();
+        logger.info("Scheduled GitHub data sync completed.");
     }
 
-    private void syncAllRepositories() {
+    private void syncData() {
+        int successfullySyncedRepositories = 0;
         for (String repositoryName : repositoriesToMonitor) {
             try {
-                dataSyncService.syncData(repositoryName);
+                dataSyncService.syncRepository(repositoryName);
                 logger.info("GitHub data sync completed successfully for repository: " + repositoryName);
```

Comment:
successfullySyncedRepositories++ here as well?

Reply:
Could you elaborate further on what you would have changed here?

Reply:
Wouldn’t it have to be increased after the sync is successful here as well? Right now it’s only increased in case of a IOexception.

Reply:
Yes you’re totally right😂

Reply:
I'm a bit confused also, why are those changes even displayed? I added them to `develop`, but I guess it does not matter.

Reply:
If I had to guess, it's because of how the rebase got handled by Git since the branch was created before you pushed the commits on `develop`. If you look at the commit history of the branch, it displays your raw commits from `develop` in there.